api/client/build.go

```diff
@@ -168,8 +181,12 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	// Dockerfile which uses trusted pulls.
 	context = replaceDockerfileTarWrapper(context, newDockerfile, relDockerfile)
 
+	// clear the progress buffer if -q is provided.
```

Comment:
Since we have access to `suppressOutput`, I'd prefer to check that instead of checking the type of `progBuff` - feels a bit risky and indirect to me.